users/arxiv/users/legacy/accounts.py

```diff
@@ -0,0 +1,257 @@
+        endorsements=endorsements.get_endorsements(user)
+    )
+    return user, auths
+
+
+def _update_field(to_update: Any, update_with: Any) -> None:
+    if to_update != update_with:
+        to_update = update_with
+
+
+def _get_user_data(user_id: str) -> Tuple[DBUser, DBUserNickname, DBProfile]:
+    with util.transaction() as session:
+        db_user, db_nick, db_profile = (
+            session.query(DBUser, DBUserNickname, DBProfile)
+            .filter(DBUser.user_id == user_id)
+            .filter(DBUser.flag_approved == 1)
+            .filter(DBUser.flag_deleted == 0)
+            .filter(DBUser.flag_banned == 0)
+            .filter(DBUserNickname.flag_primary == 1)
+            .filter(DBUserNickname.flag_valid == 1)
+            .filter(DBUserNickname.user_id == DBUser.user_id)
+            .filter(DBProfile.user_id == DBUser.user_id)
+            .first()
+        )
+        if not db_user:
+            raise exceptions.NoSuchUser('User does not exist')
+    return db_user, db_nick, db_profile
+
+
+def _create(user: domain.User, password: str, ip: str, remote_host: str) \
+        -> Tuple[DBUser, DBUserNickname, DBProfile]:
+    if user.name is None:
+        raise ValueError('User name must be set')
+    if user.profile is None:
+        raise ValueError('User profile must be set')
+
+    with util.transaction() as session:
+        # Main user entry.
+        db_user = DBUser(
+            first_name=user.name.forename,
+            last_name=user.name.surname,
+            suffix_name=user.name.suffix,
+            email=user.email,
+            policy_class=DBPolicyClass.PUBLIC_USER,
+            joined_ip_num=ip,
+            joined_remote_host=remote_host,
+            joined_date=util.now(),
+            tracking_cookie='',  # TODO: how to set?
+        )
+        session.add(db_user)
+
+        # Nickname is (apparently) where we keep the username.
+        db_nick = DBUserNickname(
+            user=db_user,
+            nickname=user.username,
+            flag_valid=1,
+            flag_primary=1
+        )
+        session.add(db_nick)
+
+        def _has_group(group: str) -> int:
+            if user.profile is None:
+                return 0
+            return int(group in user.profile.submission_groups)
+
+        db_profile = DBProfile(
+            user=db_user,
+            country=user.profile.country,
+            affiliation=user.profile.organization,
+            url=user.profile.homepage_url,
+            rank=user.profile.rank,
+            archive=user.profile.default_archive,
+            subject_class=user.profile.default_subject,
+            original_subject_classes='',
+            flag_group_physics=_has_group('grp_physics'),
+            flag_group_math=_has_group('grp_math'),
+            flag_group_cs=_has_group('grp_cs'),
+            flag_group_q_bio=_has_group('grp_q-bio'),
+            flag_group_q_fin=_has_group('grp_q-fin'),
+            flag_group_stat=_has_group('grp_stat'),
```

Comment:
Per our earlier discussion, a reminder that `grp_eess` and `grp_econ` are missing.

Reply:
Added